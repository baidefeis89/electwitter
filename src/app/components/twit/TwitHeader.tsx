import React from 'react';

import styled from '@emotion/styled';
import ILiterals from '../../../i18n';
import { ProfileImgSmall } from '../common/profileImage';
import { Icon } from '@material-ui/core';
import VerifiedToken from './verified';
import { Subtitle } from '../common/textComponents';

const TwitTitle = styled('span')`
  font-size: 20px;
  font-weight: bold;
`;

const TwitTitleContainer = styled('div')`
  display: flex;
  margin-left: 5px;
  flex-direction: column;
`;

const TwitHeaderContainer = styled('div')`
  display: flex;
  align-items: center;
`;

const RtInfo = styled('div')`
  color: #5b7086;
  font-weight: bold;
  margin-bottom: 10px;
  .material-icons {
    font-size: 1rem;
  }
`;

interface ITwitHeader {
  literals: ILiterals,
  img: string,
  name: string,
  username: string,
  retwit: boolean,
  verified: boolean,
  rtInfo?: string
}
const TwitHeader = ({literals, img, name, username, retwit, verified, rtInfo}: ITwitHeader) => {
  return (
    <>
      {retwit && <RtInfo><Icon>repeat</Icon> {rtInfo} {literals.RETWEETED_LBL}</RtInfo>}
      <TwitHeaderContainer>
        <ProfileImgSmall src={img} />
        <TwitTitleContainer>
        <TwitTitle>
          {name}{verified && <VerifiedToken />}
        </TwitTitle>
        <Subtitle>@{username}</Subtitle>
        </TwitTitleContainer>
      </TwitHeaderContainer>
    </>
  );
}

export default TwitHeader;