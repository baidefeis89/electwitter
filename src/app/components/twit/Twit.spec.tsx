import React from 'react';

import TwitHeader from './TwitHeader';
import { render } from '@testing-library/react';

describe('Twit Card', () => {
  const literals = {};
  it('should render', () => {
    const { container } = render(
      <TwitHeader
        literals={literals}
        img={'https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/golden-retriever-royalty-free-image-506756303-1560962726.jpg?crop=0.672xw:1.00xh;0.166xw,0&resize=640:*'}
        name={'Real name'}
        username={'User name'}
        retwit={false}
        verified={false}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

});
