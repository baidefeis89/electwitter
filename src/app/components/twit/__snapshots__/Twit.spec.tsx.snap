// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Twit Card should render 1`] = `
<div
  class="css-1sg2lsz"
>
  <img
    class="css-15euz30"
    src="https://hips.hearstapps.com/hmg-prod.s3.amazonaws.com/images/golden-retriever-royalty-free-image-506756303-1560962726.jpg?crop=0.672xw:1.00xh;0.166xw,0&resize=640:*"
  />
  <div
    class="css-swff5h"
  >
    <span
      class="css-1aylpg"
    >
      Real name
    </span>
    <span
      class="css-15c2s3j"
    >
      @
      User name
    </span>
  </div>
</div>
`;
